src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -257,5 +257,42 @@ internal static CSharpType FromSystemType(BuildContext context, Type type, IEnum
             provider = this.Implementation as T;
             return provider != null;
         }
+
+        /// <summary>
+        /// Check whether two CSharpType instances equal or not
+        /// This is not the same as left.Equals(right) because this function only checks the names
+        /// </summary>
+        /// <param name="left"></param>
+        /// <param name="other"></param>
+        /// <returns></returns>
+        public bool EqualsByName(CSharpType? other)
+        {
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            if (this is null || other is null)
+            {
+                return false;
+            }
+
+            if (Namespace != other.Namespace)
+                return false;
+
+            if (Name != other.Name)
+                return false;
+
+            if (Arguments.Length != other.Arguments.Length)
+                return false;
+
+            for (int i = 0; i < Arguments.Length; i++)
+            {
+                if (Arguments[i].Name != other.Arguments[i].Name)
+                    return false;
+            }
```

Comment:
should we here recursively check `EqualsByName` on all the arguments? Here we only check their names.

Reply:
nice catch, updated.